controller/destination/listener.go

```diff
@@ -85,9 +86,13 @@ func (l *endpointListener) toWeightedAddrSet(endpoints []common.TcpAddress) *pb.
 			if len(resultingPods) == 0 || resultingPods[0] == nil {
 				log.Errorf("Could not find pod for IP [%s], this IP will be sent with no metric labels.", ipAsString)
```

Comment:
this raises some eyebrows... It seems like a Real Problem if we're returning pods that we don't know about.

Reply:
yeah, i'm not sure if this could ever actually happen. filed #783 to track further investigation.